src/main/java/nl/tudelft/cse1110/andy/AndyOnWebLab.java

```diff
@@ -23,7 +24,7 @@ public static void main(String[] args) {
         if (workDir == null) { System.out.println("No WORKING_DIR environment variable."); System.exit(-1); }
         if (outputDir == null) { System.out.println("No OUTPUT_DIR environment variable.");  System.exit(-1); }
 
-        WebLabResultWriter writer = new WebLabResultWriter(new RandomAsciiArtGenerator());
+        WebLabResultWriter writer = new WebLabResultWriter(PropertyUtils.getVersionInformation(), new RandomAsciiArtGenerator());
```

Comment:
In here, let's use the constructor that only requires the ascii generator. (This constructor calls the property version).

In general words, no client classes should need to know about this property version thingy.

Reply:
I made it so that it uses an empty constructor. Is there any point in passing the ascii generator here, it's also an internal component?

Reply:
My rule of thumb is: if the dependency is a sort of a tool (i.e., something with some logic in it), like the case of the `RandomAsciiArtGenerator`, I pass it via the constructor, and I do not mind requiring clients to instantiate it and pass it over. This reminds clients of the dependencies the class has.

Now, the version information is just "some information"; it's not a component per se. We will never replace it by something else. If we replaced it by simple static method calls, and no dependencies, we'd be happy too. We'd require the test to have the same .properties file in the resources folder, that's not a big deal. Passing it as a dependency enables us to mock, which is also good and clean, but then, I'd make sure clients are not required to pass it when instantiating the main class.

Reading suggestion: Growing Object-Oriented Systems Guided by Tests, chapter 6!